packages/caravan-health/src/waste.ts

```diff
@@ -161,17 +165,29 @@ export class WasteMetrics extends WalletMetrics {
 
     let vsize: number;
     if (scriptType === "P2SH") {
-      vsize = 276;
+      const signatureLength = 72 + 1; // approx including push byte
+      const keylength = 33 + 1; // push byte
+      vsize = signatureLength * config.m + keylength * config.n;
     } else if (scriptType === "P2WSH") {
-      vsize = 105.75;
+      let total = 0;
+      total += 1; // segwit marker
+      total += 1; // segwit flag
+      total += getWitnessSize(config.m, config.n); // add witness for each input
```

Comment:
```
// add witness for each input
```
this is probably just copy pasta right? doesn't make sense in this context.

Reply:
Yes